src/test/kotlin/io/perfometer/dsl/HttpDslSpecification.kt

```diff
@@ -107,10 +107,28 @@ internal class HttpDslSpecification {
         }.build()
 
         val securedRequestsCount = securedScenario.steps.filterIsInstance<RequestStep>()
-                .mapNotNull { it.request.authorization }
-                .filter { authHeader -> authHeader.first == "Authorization" && authHeader.second == "Basic $credentialsEncoded" }
+                .flatMap { it.request.headers().entries }
+                .filter { header -> header.key == "Authorization"  && header.value == "Basic $credentialsEncoded" }
```

Comment:
Consider creating an object for holding commonly used HTTP Headers, like `Authorization`, instead of using String literals, e.g.

object HttpHeaders {
  const val AUTHORIZATION = "Authorization"
}

Reply:
done